### PR TITLE
Improve-Dash-feel-changing-how-it-is-interpolated-currently-linear

### PR DESCRIPTION
**Pull request by EdwardRegasDuncan**
Updated how dash is interpolated using Ease Out

### CODE REVIEW

**Comment by EdwardRegasDuncan** [comment]
Task linked: [CU-2gbp7uh Improve Dash feel, changing how it is interpolated (currently linear)](https://app.clickup.com/t/2gbp7uh)

**Review by Bernatmago — APPROVED** [review]

File: Gameplay/entities/player/PlayerController.h [hunk 1]
```diff
@@ -147,6 +147,7 @@ class PlayerController : public Script
 	SERIALIZE_FIELD(float, _dash_duration);
 	SERIALIZE_FIELD(float, _dash_distance);
 	SERIALIZE_FIELD(float, _dash_cooldown);
+	SERIALIZE_FIELD(int, _dash_scaler);
```
Comment (monikamnc) [inline Gameplay/entities/player/PlayerController.h:150]:
```suggestion
	SERIALIZE_FIELD(unsigned int, _dash_scaler);
```
Hii! :3
I was testing the PR and when you add a negative number it brokes the position of the player game object and the camera look at too. I know it's not supposed to be negative in any case but I think we can change the _dash_scaler to unsigned int to be safe, or what do you think?
image

**Review by monikamnc — CHANGES_REQUESTED** [review]

File: Gameplay/entities/player/PlayerController.cpp [hunk 2]
```diff
@@ -601,10 +602,15 @@ void Hachiko::Scripting::PlayerController::DashController()
 
 	_dash_progress += Time::DeltaTime() / _dash_duration;
 	_dash_progress = _dash_progress > 1.0f ? 1.0f : _dash_progress;
+	
+	// using y = x^p
+	float acceleration = 1.0f - math::Pow((1.0f - _dash_progress) / 1.0f, _dash_scaler);
+	
 
 	// TODO: Instead of approaching to _dash_end linearly, dash must have some sort
```
Comment (monikamnc) [inline Gameplay/entities/player/PlayerController.cpp:610]:
```suggestion
	
```
Erase the TODO if it's finished ^-^

File: Gameplay/entities/player/PlayerController.cpp [hunk 3]
```diff
@@ -601,10 +602,15 @@ void Hachiko::Scripting::PlayerController::DashController()
...
 	_dash_progress += Time::DeltaTime() / _dash_duration;
 	_dash_progress = _dash_progress > 1.0f ? 1.0f : _dash_progress;
+	
+	// using y = x^p
+	float acceleration = 1.0f - math::Pow((1.0f - _dash_progress) / 1.0f, _dash_scaler);
+	
 
-	// TODO: Instead of approaching to _dash_end linearly, dash must have some sort
+	
 	// of an acceleration.
```
Comment (monikamnc) [inline Gameplay/entities/player/PlayerController.cpp:611]:
```suggestion
	
```

**Review by monikamnc — APPROVED** [review]
Thanks for your work ^-^

**Pull request merged**